arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArLocationDataSource.kt

```diff
@@ -0,0 +1,584 @@
+        locationManager?.removeUpdates(internalLocationListener)
+        internalLocationListener = null
+
+        // stop update heading if it is started.
+        stopUpdateHeading()
+    }
+
+    /**
+     * Validates [minTime] and [minDistance].
+     *
+     * @since 100.6.0
+     */
+    private fun checkTimeDistanceParameters(minTime: Long, minDistance: Float) {
+        if (minTime < 0) {
+            throw IllegalArgumentException(String.format(PARAMETER_OUT_OF_BOUNDS_MSG, "minTime"))
+        }
+        minimumUpdateTime = minTime
+
+        if (minDistance < 0) {
+            throw IllegalArgumentException(
+                String.format(
+                    PARAMETER_OUT_OF_BOUNDS_MSG,
+                    "minDistance"
+                )
+            )
+        }
+        minimumUpdateDistance = minDistance
+    }
+
+    /**
+     * Select the default location providers, network will be used first if it is available.
+     *
+     * @since 100.6.0
+     */
+    private fun selectProvidersByDefault() {
+        locationManager?.let {
+            // Check if the network location service enabled
+            if (it.isProviderEnabled(LocationManager.NETWORK_PROVIDER)) {
+                selectedLocationProviders.add(LocationManager.NETWORK_PROVIDER)
+            }
+
+            // check if the GPS location service enabled,if true, use it too.
+            if (it.isProviderEnabled(LocationManager.GPS_PROVIDER)) {
+                selectedLocationProviders.add(LocationManager.GPS_PROVIDER)
+            }
+        }
+    }
+
+    /**
+     * Called when the AutoPanMode is changed on the associated LocationDisplay, this method populates the heading
+     * [LocationDisplay.getHeading] on the associated LocationDisplay if the AutoPanMode is changed to
+     * [AutoPanMode.COMPASS_NAVIGATION].
+     *
+     * @since 100.6.0
+     */
+    override fun onAutoPanModeChanged(autoPanModeEvent: LocationDisplay.AutoPanModeChangedEvent) {
+        autoPanMode = autoPanModeEvent.autoPanMode
+        if (!isStarted) {
+            return
+        }
+    }
+
+    /**
+     * Starts the location provider.
+     *
+     * @since 100.6.0
+     */
+    @SuppressLint("MissingPermission")
+    private fun startLocationProviders() {
+        if (internalLocationListener == null) {
+            internalLocationListener = InternalLocationListener()
+        }
+
+        for (provider in selectedLocationProviders) {
+            locationManager?.requestLocationUpdates(
+                provider,
+                minimumUpdateTime,
+                minimumUpdateDistance,
+                internalLocationListener
+            )
+        }
+    }
+
+    /**
+     * Updates the CoreLocationDataSource location with the provided [location], indicating if the provided location is
+     * the [lastKnown] location.
+     *
+     * @since 100.6.0
+     */
+    private fun updateCoreLocation(location: android.location.Location?, lastKnown: Boolean) {
```

Comment:
I think we should rename the method here in this context, as our users don't know the concept of "Core". I suggest, just call it `updateEsriLocation()`. The doc would also have to be changed to avoid the term "Core".